src/lib/containers/discussion_forum/DiscussionThread.tsx

```diff
@@ -0,0 +1,173 @@
+import moment from 'moment'
+import React, { useEffect, useState } from 'react'
+import { formatDate } from '../../utils/functions/DateFormatter'
+import {
+  useGetRepliesInfinite,
+  useGetThread,
+} from '../../utils/hooks/SynapseAPI/forum/useForum'
+import { SynapseClient } from '../../utils'
+import { SMALL_USER_CARD } from '../../utils/SynapseConstants'
+import { useSynapseContext } from '../../utils/SynapseContext'
+import {
+  Subscription,
+  SubscriptionObjectType,
+  Topic,
+} from '../../utils/synapseTypes/Subscription'
+import Typography from '../../utils/typography/Typography'
+import { getMessage } from '../DiscussionSearchResult'
+import UserCard from '../UserCard'
+import { displayToast } from '../ToastMessage'
+import { DiscussionReply } from './DiscussionReply'
+import { Button, FormControl } from 'react-bootstrap'
+import IconSvg from '../IconSvg'
+import Tooltip from '../../utils/tooltip/Tooltip'
+import MarkdownSynapse from '../markdown/MarkdownSynapse'
+import { ObjectType } from '../../utils/synapseTypes'
+import { getSubscription } from '../../utils/functions/getSubscription'
+import {
+  useDeleteSubscription,
+  usePostSubscription,
+} from '../../utils/hooks/SynapseAPI/subscription/useSubscription'
+
+export type DiscussionThreadProps = {
+  threadId: string
+  limit: number
+}
+
+const FOLLOWING_TEXT = 'You are following this topic. Click to stop following.'
+const UNFOLLOWING_TEXT = 'You are not following this topic. Click to follow.'
+
+export function DiscussionThread(props: DiscussionThreadProps) {
+  const { threadId, limit } = props
+
+  const { accessToken } = useSynapseContext()
+  const [orderByDatePosted, setOrderByDatePosted] = useState(true)
+  const [message, setMessage] = useState<string>()
+  const [subscribed, setSubscribed] = useState<Subscription>()
+  const [isLoading, setIsLoading] = useState(false)
+
+  const { data: threadData } = useGetThread(threadId)
+  const { mutate: updateSubscription } = usePostSubscription()
+  const { mutate: deleteSubscription } = useDeleteSubscription()
+
+  useEffect(() => {
+    getSubscription(accessToken, threadId, SubscriptionObjectType.THREAD).then(
+      result => setSubscribed(result),
+    )
+  }, [accessToken, threadId, handleFollowBtn])
```

Comment:
I tried using react-query, but when using it returns an empty array. So ended up using the useEffect and useState.